sdk/eventhub/azure-eventhub/azure/eventhub/_buffered_producer/_buffered_producer_dispatcher.py

```diff
@@ -51,7 +51,9 @@ def __init__(
         self._amqp_transport = amqp_transport
 
         if not executor:
-            self._executor = ThreadPoolExecutor()
+            # set max workers to the number of partitions & one for the close operation.
+            max_workers: int = len(self._partition_ids) + 1
+            self._executor = ThreadPoolExecutor(max_workers=max_workers)
```

Comment:
are there any tests we can add?

Reply:
also, does async need to be updated too?

Reply:
I didnt add any tests because I would just be checking if a private variable is set to the count.

Reply:
> also, does async need to be updated too?

async fires off futures to handle this, so it works similar to this and no ThreadPool usage over there.